src/main/python/modelers/regression_models.py

```diff
@@ -0,0 +1,122 @@
+import numpy as np
+import pandas as pd
+from argparse import Namespace
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LinearRegression
+from sklearn.neural_network import MLPRegressor
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.metrics import r2_score
+from sklearn.metrics import mean_squared_error
+
+
+class TweetsModeler:
+    def __init__(self, args):
+        self.dfWithFeatures = args.dfWithFeatures
+        self.featureColumnIndexes = args.featureColumnIndexes
+        self.points = args.points
+        self.regressionMethod = args.regressionMethod
+        self.regResultsDF = pd.DataFrame()
+        self.resultsDict = dict()
+
+    @classmethod
+    def getModelResutls(cls, dfWithFeatures, featureColumnIndexes, points, regressionMethod):
+        """
+        Returns an tweetsModeler object. the object has two important attributes, regResutlsDF and resultsDict
+
+        @param dfWithFeatures: The data frame that will have the features from an analysis
+        @param featureColumnIndexes: A list of the indexes that contain the feature columns
+        @param points: True if points should be predicted, false if accuracy should be predicted.
+        @param regressionMethod: An integer value signalling which regression method to be used
+        @return: An object with attributes for predictions dataframe and regression results
+        @rtype: object
+        """
+        args = Namespace(
```

Comment:
I've never seen this, that's interesting. Your PR is helping me understand how Python uses the Factory design pattern. I'm surprised you used this pattern for the class.